src/app/hooks/useExploreServers.ts

```diff
@@ -0,0 +1,38 @@
+import { AccountDataEvent } from '../../types/matrix/accountData';
+import { useAccountData } from './useAccountData';
+import { useMatrixClient } from './useMatrixClient';
+
+export type InCinnyExploreServersContent = {
+  servers?: string[];
+};
+
+export type ExploreServerListAction =
+  | {
+      type: 'APPEND';
+      server: string;
+    }
+  | {
+      type: 'DELETE';
+      server: string;
+    };
+
+export const useExploreServers = (): [string[], (action: ExploreServerListAction) => void] => {
+  const mx = useMatrixClient();
+  const userAddedServers =
+    useAccountData(AccountDataEvent.CinnyExploreServers)?.getContent<InCinnyExploreServersContent>()
+      ?.servers ?? [];
+
+  const setUserAddedServers = (action: ExploreServerListAction) => {
```

Comment:
instead of using the action we should make two separate functions i.e. `addServer` and `removeServer`.

Reply:
those should also return a promise so button click can show the spinner animation while the request complete.